### PR TITLE
Forecasting docs enhancements

### LINKED ISSUE #82: Review forecast proposal / snapshot endpoint documentation for clarity and correctness

### PR DESCRIPTION
**Pull request by catkins-miso**
The primary updates are:
* added `proposal-begins` in forecast proposal header
* expanded concepts
* extended forecast submittal documentation to describe the "partial success" paradigm

### CODE REVIEW

File: docs/_data/components/schemas/headers.yaml
Comment (caindy) [inline docs/_data/components/schemas/headers.yaml]: @getorymckeag This is an important change, and I wanted to get your 👀 on this.

File: docs/articles/forecast-windows.md
Comment (caindy) [inline docs/articles/forecast-windows.md:49]: @getorymckeag This describes how the new header field is used to validate the proposal. There was some ambiguity that comes up with allowing some individual resource forecast proposals to be invalid, and this header property makes the intent clear--all resource proposals in the PATCH should start at `proposal-begins`.

File: docs/concepts.md
Comment (caindy) [inline docs/concepts.md:52]: Is it worth mentioning that a clearinghouse could be hosted by an entity other than a Transmission Provider or Reliability Coordinator? For example, a TOP could clear Ratings on a facility from multiple TOs, acting as the sole Ratings Provider on that facility from the TP's perspective. TO -> TOP's TROLIE -> RC's TROLIE

File: docs/articles/forecast-windows.md
Comment (getorymckeag) [reply docs/articles/forecast-windows.md:49]: So... devil's advocate here.  Without having the value, you can determine this implicitly simply by looking at the earliest period in the PATCH request body.  If you add the value, then the server must implement two validations:
1.  Enforce the invariant that all period starts are >= proposal-begins.
2.  What does the system do if there is a gap between "proposal-begins" and the first period in the PATCH?  Is it still valid?  What should the server do if "proposal-begins" is late, but none of the actual periods are?  

I would think that these discussions are unavoidable, and certainly represent work.  So, what's the tradeoff?  What do we gain from having it be explicitly called out in the header that is worth the work we've created?

File: docs/concepts.md
Comment (getorymckeag) [reply docs/concepts.md:52]: I think this is good enough really.  TOPs that actually want to do that will be smart enough to figure out, and those that aren't are just as likely to be confused by the statement as get any value from it.

File: docs/example-narratives/submitting-forecasts.md
Comment (getorymckeag) [inline docs/example-narratives/submitting-forecasts.md]: "an" ratings forecast?

File: docs/example-narratives/submitting-forecasts.md
Comment (getorymckeag) [inline docs/example-narratives/submitting-forecasts.md]: The way you've phrased this, the whole paragraph belongs in an ADR, not in the examples.  I've learned this from some of our doc writing people, who have complained to me about architects writing install or usage guides.  The people reading the examples sections really don't care about why it works the way it does, they only need to know how.  It's too much.  

You're trying to set up for your example that shows it is possible to break up the proposal into chunks- I would consider rephrasing it to just illustrate its something you can do and how, not why they should do it or why TROLIE supports it.

File: docs/images/forecast-states.puml
Comment (getorymckeag) [inline docs/images/forecast-states.puml:1]: I think this will be helpful for a lot of people.

**Review by getorymckeag — CHANGES_REQUESTED** [review]
As usual, you're a fantastic writer.  But yes, I do have some requests, and I think you need to sell me a little harder on the `proposal-begins`  field.

File: docs/articles/forecast-windows.md
Comment (caindy) [reply docs/articles/forecast-windows.md:49]: If I send you a forecast proposal just after a deadline has past but some of the forecast start on the hour appropriate to the current forecast window and others start in the prior hour--what was my intent?

If the header makes that explicit, there's no ambiguity.

Re: the validations, the first is a check we're doing already since `proposal-begins == forecast-window-first-hour` and the later is checked against each resource forecast to determine if it is late. For second point, the whole idea with this is that we need a way for the provider to be unambiguous about which Forecast Window they're targeting.

File: docs/example-narratives/submitting-forecasts.md
Comment (caindy) [inline docs/example-narratives/submitting-forecasts.md]:
```suggestion
a Ratings Forecast to their Transmission Provider. TROLIE provides the
```

File: docs/articles/forecast-windows.md
Comment (caindy) [reply docs/articles/forecast-windows.md:49]: Please note I renamed this to `begins` from `proposal-begins` (cherry-picked from #89) because the way this is factored, that field shows up in all headers (request/response, RT/forecast). That was intentional--they all benefit from the explicit identification of the operational window--but the "proposal" part clearly doesn't make sense in all contexts.

**Comment by caindy** [comment]
> As usual, you're a fantastic writer. But yes, I do have some requests, and I think you need to sell me a little harder on the `proposal-begins` field.

Thanks for the review and really good feedback. I've implemented the changes you requested and replied with my sales pitch.

**Comment by caindy** [comment]
Just a note about the broken build--the Spectral linter is wrong here, it's just not setup to look through the `allOf` attributes that are brought in.
image
I'll try to figure a way to keep this DRY yet make the linter happy.

File: docs/articles/forecast-windows.md
Comment (getorymckeag) [reply docs/articles/forecast-windows.md:49]: > If I send you a forecast proposal just after a deadline has past but some of the forecast start on the hour appropriate to the current forecast window and others start in the prior hour--what was my intent?
> 
> If the header makes that explicit, there's no ambiguity.
> 
> Re: the validations, the first is a check we're doing already since `proposal-begins == forecast-window-first-hour` and the later is checked against each resource forecast to determine if it is late. For second point, the whole idea with this is that we need a way for the provider to be unambiguous about which Forecast Window they're targeting.

I get it now.  Sold.

**Review by getorymckeag — APPROVED** [review]

**Comment by getorymckeag** [comment]
> Just a note about the broken build--the Spectral linter is wrong here, it's just not setup to look through the `allOf` attributes that are brought in. image I'll try to figure a way to keep this DRY yet make the linter happy.

Approved, I'm guessing you want to make the linter happy before merging?  I'll re-approve after the cosmetics are handled.

**Pull request merged**
